### ISSUE TITLE
Metrics remain zero when training with YOLOv8-P2 model

### ISSUE THREAD

**Issue by daocunyang** [ultralytics/ultralytics#5951]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello, I have a custom dataset where I hope to detect both large objects and small objects, I have trained a regular yolov8s model and now want to improve the detection results on small objects. So I decided to try out the YOLOv8-P2 variant. But having trained the P2 model for a few epochs, I noticed the metrics "P, R, mAP50, mAP50-95" remained zero without any sign of model improvement. So I wonder if there is anything wrong with the way I'm using the p2 model?

I made a copy of `yolov8-p2.yaml` and specified the following:
```
nc: 10
scales: 
  s: [0.33, 0.50, 1024]

# The rest are copied exactly from yolov8-p2.yaml
backbone:
  # [from, repeats, module, args]
  - [-1, 1, Conv, [64, 3, 2]]  # 0-P1/2
  - [-1, 1, Conv, [128, 3, 2]]  # 1-P2/4
  - [-1, 3, C2f, [128, True]]
  - [-1, 1, Conv, [256, 3, 2]]  # 3-P3/8
  - [-1, 6, C2f, [256, True]]
  - [-1, 1, Conv, [512, 3, 2]]  # 5-P4/16
  - [-1, 6, C2f, [512, True]]
  - [-1, 1, Conv, [1024, 3, 2]]  # 7-P5/32
  - [-1, 3, C2f, [1024, True]]
  - [-1, 1, SPPF, [1024, 5]]  # 9

# YOLOv8.0-p2 head
head:
  - [-1, 1, nn.Upsample, [None, 2, 'nearest']]
  - [[-1, 6], 1, Concat, [1]]  # cat backbone P4
  - [-1, 3, C2f, [512]]  # 12

  - [-1, 1, nn.Upsample, [None, 2, 'nearest']]
  - [[-1, 4], 1, Concat, [1]]  # cat backbone P3
  - [-1, 3, C2f, [256]]  # 15 (P3/8-small)

  - [-1, 1, nn.Upsample, [None, 2, 'nearest']]
  - [[-1, 2], 1, Concat, [1]]  # cat backbone P2
  - [-1, 3, C2f, [128]]  # 18 (P2/4-xsmall)

  - [-1, 1, Conv, [128, 3, 2]]
  - [[-1, 15], 1, Concat, [1]]  # cat head P3
  - [-1, 3, C2f, [256]]  # 21 (P3/8-small)

  - [-1, 1, Conv, [256, 3, 2]]
  - [[-1, 12], 1, Concat, [1]]  # cat head P4
  - [-1, 3, C2f, [512]]  # 24 (P4/16-medium)

  - [-1, 1, Conv, [512, 3, 2]]
  - [[-1, 9], 1, Concat, [1]]  # cat head P5
  - [-1, 3, C2f, [1024]]  # 27 (P5/32-large)

  - [[18, 21, 24, 27], 1, Detect, [nc]]  # Detect(P2, P3, P4, P5)
```
Then I load the model using:
`model = YOLO('yolov8-p2.yaml').load('yolov8s.pt')`
and started training. 

Any insight and help is appreciated, thanks in advance

**Comment by EtAiorsW** [comment]
Hi, I have the same question, I want to improve the detection performance of small objects, I trained yolov5l-p2 and yolov8-p2 on my dataset, after 600 epochs, mAP50 and mAP50-95 is still less than 0.01, precision and recall is around 0.2. But yolov5l converged normally.

**Comment by glenn-jocher** [comment]
@EtAiorsW hi, thanks for utilizing YOLOv8 for object detection tasks. Given your description, it seems that mAP and precision-recall metrics for the YOLOv8-P2 model are displaying a lack of convergence after 600 epochs. This may be due to several reasons:

1. **Dataset characteristics**: Depending on the complexity of the dataset and the specific variations of the classes, some models may potentially achieve better results than others. The characteristics inherent to the dataset might be playing a crucial role here. YOLOv8-P2 is designed with a focus towards detecting smaller objects; if your dataset has very unique or specific small object characteristics, it might be more challenging to achieve optimal performance.

2. **Initialization weights**: The starting weights can greatly influence the performance of a model. For YOLOv8-P2, ensure that you're correctly initializing the model with the provided pre-trained weights.

3. **Hyperparameters**: Hyperparameter settings such as the learning rate, number of epochs, optimizer used, and more, can have a major impact on the model's performance. Make sure the hyperparameters are well-tuned for your specific use-case. Also, be aware that different architectures might require different sets of optimal hyperparameters.

Remember that model training involves a lot of experimentation, and you might not always see an improvement when switching architectures, especially when the change is motivated by specific object scale preferences such as small object detection in your case. 

I suggest you keep experimenting with different strategies and monitor for improvements. I hope this helps! Let me know if you have any more questions.

**Comment by daocunyang** [comment]
It turned out that I just need to train for a few more epochs, then the metrics looked correct.

**Comment by glenn-jocher** [comment]
@daocunyang that's great to hear! Sometimes it just takes a bit more time for the models to learn and start converging. Patience and careful monitoring during the training process are key. If you have any more questions or require further assistance with your models in the future, don't hesitate to reach out. Good luck with your project!
